carbon-design-system/carbon
# refactor(OverflowMenu): ariaLabel to aria-label

#12802 

Update `OverflowMenu` to use the proper `aria-label`

#### Changelog

**New**

- n/a

**Changed**

1. added `deprecated()` to the old propType
2. added new usage for `aria-label`
3. kept backwards compatibility so that both aria versions can work with `ariaLabel` (deprecated) and `aria-label` (new).

**Removed**

- n/a

#### Testing / Reviewing

1. Open Storybook > OverflowMenu in browser for Dev Tool testing
2. Open `OverflowMenu.stories.js` in code
3. Add one aria label type at a time to the component in story code. 
 - 1. Only one version can be active at a time.
   - `aria-label` (new)
   - `ariaLabel` (deprecated))
 - 2. Be sure to find the correct usage for the given component as explained in code or on the story itself
 - 3. Test examples below. 
   
   - Note: For my testing, I duplicate the lines and then comment out the one I am not using. The image blow does NOT yet comment out one of the test lines:
![image](https://user-images.githubusercontent.com/9935383/221902620-25833bd2-c904-4720-80af-658363aca1d0.png)
5. confirm that both propTypes still display in Dev Tools as intended
6. confirm that Storybook > Docs > Component API contains both propTypes as options
